llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -511,16 +511,27 @@ bool RISCVRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
       // We can encode an add with 12 bit signed immediate in the immediate
       // operand of our user instruction.  As a result, the remaining
       // offset can by construction, at worst, a LUI and a ADD.
+
+      // Special case, try to split the offset across two ADDIs.
+      uint64_t ValSubLo12 = (uint64_t)Val - (uint64_t)Lo12;
```

Comment:
Prior comment re: alignment remains unaddressed.

Reply:
The argument of adjustReg `MaybeAlign RequiredAlign` is std::nullopt, so there maybe no need to compute alignment here.
https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp#L527

Reply:
The other code needs to manage alignment because we're never allowed to misalign the stack pointer. I don't think that applies here

Reply:
Try to realignment.